web/src/widgetize.ts

```diff
@@ -82,28 +82,28 @@ export function activate(in_opts: WidgetOptions): void {
     // the HTML that will eventually go in this div.
     const $widget_elem = $("<div>").addClass("widget-content");
 
-    widgets.get(widget_type)!.activate({
+    const event_handler = widgets.get(widget_type)!.activate({
         $elem: $widget_elem,
         callback,
         message,
         extra_data,
     });
 
-    widget_contents.set(message.id, $widget_elem);
+    widget_event_handlers.set(message.id, event_handler);
     set_widget_in_message($row, $widget_elem);
 
     // Replay any events that already happened.  (This is common
     // when you narrow to a message after other users have already
     // interacted with it.)
     if (events.length > 0) {
-        $widget_elem.handle_events(events);
+        event_handler(events);
     }
 }
 
 export function handle_event(widget_event: Event & {message_id: number}): void {
-    const $widget_elem = widget_contents.get(widget_event.message_id);
+    const event_handler = widget_event_handlers.get(widget_event.message_id);
 
-    if (!$widget_elem) {
+    if (!event_handler || !message_lists.current?.get_row(widget_event.message_id)) {
         // It is common for submessage events to arrive on
         // messages that we don't yet have in view. We
         // just ignore them completely here.
```

Comment:
Just to check -- if we have a rendered feed in the background, this won't process submessages on it, but it's OK because we'll just do those when switching to the view anyway, right?

Reply:
Yep, the widgets are activated whenever views are switched, even if to one rendered in the background.

While testing for this, I realised there was a 0 consequence bug in how the current view's widgets were checked here, due to quirks in `get_row` (it returns an empty jquery object instead of undefined for legacy reasons), so have fixed that too.